LongWarOfTheChosen/Src/LW_PerkPack_Integrated/Classes/X2Condition_RevivalProtocol_LW.uc

```diff
@@ -0,0 +1,22 @@
+//---------------------------------------------------------------------------------------
+//  FILE:    X2Condition_RevivalProtocol_LW.uc
+//  AUTHOR:  Peter ledbrook
+//  PURPOSE: Allows Revival Protocol to also target stunned units.
+//---------------------------------------------------------------------------------------
+class X2Condition_RevivalProtocol_LW extends X2Condition_RevivalProtocol;
+
+event name CallMeetsCondition(XComGameState_BaseObject kTarget)
+{
+	local XComGameState_Unit TargetUnit;
+    local name Result;
+
+	TargetUnit = XComGameState_Unit(kTarget);
+
+    Result = super.CallMeetsCondition(kTarget);
+    if (Result == 'AA_UnitIsNotImpaired' && TargetUnit != none && TargetUnit.IsStunned())
```

Comment:
It might be better at this point to put both effect conditions and effects to be removed in config arrays (and then check if unit is affected by the effect name in the array), in case we want to change which effects it works on (for example I wanted to actually remove daze from it for a while), or just to make it easier to make it work with other mod effects (hypothetically something like heavy daze)

Reply:
Eh. Thought about it, but it's more effort than I'm willing to invest right now. If a configurable array is warranted at a later date, we can do it. But you also have to deal with the `X2Effect_RemoveEffects` array.